### PR TITLE
Feature/DIOS-7069 Better Layer Switching

### PR DESCRIPTION
**Pull request by mupatt-dolby**
https://jira.dolby.net/jira/browse/DIOS-7069 

Separate disposable effects for pause and playing. Don't re-update pending quality

### CODE REVIEW

File: app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingScreen.kt [hunk 1]
```diff
@@ -58,17 +59,12 @@ fun StreamingScreen(viewModel: StreamingViewModel = hiltViewModel(), onBack: ()
             }
 
             uiState.subscribed && uiState.error == null && uiState.videoTrack != null -> {
-                DisposableEffect(uiState.videoTrack, uiState.pendingSelectedStreamQualityType) {
+                DisposableEffect(Unit) {
```
Comment (mupatt-dolby) [inline app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingScreen.kt:62]: Always add a pause effect regardless of state

File: app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingScreen.kt [hunk 2]
```diff
@@ -80,6 +76,21 @@ fun StreamingScreen(viewModel: StreamingViewModel = hiltViewModel(), onBack: ()
                         viewModel.pauseVideo()
                     }
                 }
+                DisposableEffect(uiState.videoTrack, uiState.pendingSelectedStreamQualityType) {
```
Comment (mupatt-dolby) [inline app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingScreen.kt:79]: Add an effect for changed video tracks or selected stream quality that only does resuming

File: app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingViewModel.kt [hunk 3]
```diff
@@ -430,8 +430,7 @@ class StreamingViewModel @Inject constructor(
         streamQualityType?.let { qualityType ->
             _uiState.update {
                 it.copy(
-                    selectedStreamQualityType = qualityType,
-                    pendingSelectedStreamQualityType = null
+                    selectedStreamQualityType = qualityType
```
Comment (mupatt-dolby) [inline app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingViewModel.kt:433]: Don't call playVideo twice in a row, which happens when you null out pendingSelectedStreamQuality, which is being observed

**Review by mshaa92 — APPROVED** [review]
Looks good

**Pull request merged**